File: sorting/bubble_sort.c
#include <stdio.h>
void display_array(int arr[], int array_size);
void bubble_sort(int a[], int len);
void main()
{
    int array_size = 5;
    int a[] = {5, 4, 3, 2, 1};
    bubble_sort(a, array_size);    
    display_array(a, array_size);
}

void display_array(int arr[], int array_size){
    for (int i = 0; i < array_size; i++)
    {
        printf("%d\n", arr[i]);
    }
}
void bubble_sort(int a[], int len){
    short isSorted = 0;
    while(!isSorted){
        isSorted = 1;
        for(int i = 0; i < len; i++){
            if(a[i] > a[i+1]){
                isSorted = 0;
                int t = a[i];
                a[i] = a[i+1];
                a[i+1] = t;                
            }
        }
    }
}

File: sorting/bubble_sort.py

def bubbleSort(a):
    isSorted = False    
    while not isSorted:
        isSorted = True
        for i in range(0, len(a)-1):
            if a[i] > a[i+1]:
                isSorted = False                
                t = a[i]
                a[i] = a[i+1]
                a[i+1] = t
    return a


# bubbleSort([2,5,1,3,4])
# bubbleSort([2,1,5,3,4])
# bubbleSort([1,2,3,5,4,6,7,8])
# bubbleSort([1,2,5,3,7,8,6,4])
